sdk/docs/core/resources/telemetry_consumer.puml

```diff
@@ -0,0 +1,22 @@
+@startuml TELEMETRY_CONSUMER
+
+state TelemetryConsumer {
+
+  state Ready
+  Ready : <b>telemetry_consumer_subscribe_begin/</b> start sub timer, ^MQTT_SUB_REQ(...)
```

Comment:
I don't think so. The telemetry_consumer_subscribe_begin is going to be an api call from the application, which triggers the other events. The MQTT_SUB_REQ sent out doesn't get seen by the hfsm, so it can't act on that event. We could create another custom event just so the timer start and sending of the sub request happens within the HFSM instead of the api call, but this feels a bit unnecessarily layered to me - I'll think on it though.

Reply:
That makes sense, I was basing my comment on what you did for the RPC client's `az_mqtt5_rpc_client_subscribe_begin`

Another benefit of taking this approach is that if the HFSM goes to a faulted state and the user tries to send out a subscribe_begin we won't start a timer since the case would be handled by the default of the `faulted` state

Reply:
Good call - it would be good to not send the subscribe if we're in faulted - I didn't add this extra layer in telemetry originally since there's only one valid state and subscribe would always behave the same way in it. But in faulted it shouldn't, so I'll separate that out.

Reply:
updated in the new PR